glados-web/src/routes.rs

```diff
@@ -1352,27 +1040,28 @@ fn decouple_nodes_and_censuses(
 }
 
 pub async fn single_census_view(
-    census_id: HttpQuery<HashMap<String, String>>,
+    params: HttpQuery<HashMap<String, String>>,
     Extension(state): Extension<Arc<State>>,
 ) -> Result<HtmlTemplate<SingleCensusViewTemplate>, StatusCode> {
-    let max_census_id = match get_max_census_id(&state).await {
+    let subprotocol = get_subprotocol_from_params(&params);
+    let max_census_id = match get_max_census_id(&state, subprotocol).await {
         None => return Err(StatusCode::from_u16(404).unwrap()),
         Some(max_census_id) => max_census_id,
     };
 
-    let census_id: Option<i32> = match census_id.get("census-id") {
+    let census_id: Option<i32> = match params.get("census-id") {
         None => return Err(StatusCode::from_u16(404).unwrap()),
         Some(census_id) => match census_id.parse::<i32>() {
             Ok(census_id) => Some(census_id),
             Err(_) => Some(max_census_id.id),
         },
     };
 
-    let client_diversity_data =
-        match generate_client_diversity_data(&state, census_id, max_census_id).await {
-            None => return Err(StatusCode::from_u16(404).unwrap()),
-            Some(client_diversity_data) => client_diversity_data,
-        };
+    let client_diversity_data = match generate_client_diversity_data(&state, max_census_id.id).await
+    {
+        None => return Err(StatusCode::from_u16(404).unwrap()),
+        Some(client_diversity_data) => client_diversity_data,
+    };
```

Comment:
Doesn't this break the client graph for the single_census_view( if a only census is selected. As now we only show the latest census no matter what. I think we should show the graph from the specific audit be displayed

```suggestion
    let client_diversity_data = match generate_client_diversity_data(&state, census_id.id).await
    {
        None => return Err(StatusCode::from_u16(404).unwrap()),
        Some(client_diversity_data) => client_diversity_data,
    };
```

I think this suggestion should fix it ^

Reply:
Good catch! fixed